packages/node/src/node/Node.ts

```diff
@@ -38,6 +38,7 @@ const logger = Logger.get("Node");
 export abstract class Node<T extends Node.CommonRootEndpoint = Node.CommonRootEndpoint> extends Endpoint<T> {
     #environment: Environment;
     #runtime?: NetworkRuntime;
+    #targetState: "online" | "offline" = "offline";
```

Comment:
I think this makes sense but it should go into NodeLifecycle, that's where all the state management cruft is